egs2/wsj0-2mix/enh1/enh.sh

```diff
@@ -0,0 +1,682 @@
+    else
+       log "Skip stage 2: Speed perturbation"
+    fi
+fi
+
+if [ -n "${speed_perturb_factors}" ]; then
+    train_set="${train_set}_sp"
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+    if [ "${feats_type}" = raw ]; then
+        log "Stage 3: Format wav.scp: data/ -> ${data_feats}/org/"
+
+        # ====== Recreating "wav.scp" ======
+        # Kaldi-wav.scp, which can describe the file path with unix-pipe, like "cat /some/path |",
+        # shouldn't be used in training process.
+        # "format_wav_scp.sh" dumps such pipe-style-wav to real audio file
+        # and also it can also change the audio-format and sampling rate.
+        # If nothing is need, then format_wav_scp.sh does nothing:
+        # i.e. the input file format and rate is same as the output.
+
+        for dset in "${train_set}" "${dev_set}" ${eval_sets}; do
+            utils/copy_data_dir.sh data/"${dset}" "${data_feats}/org/${dset}"
+            rm -f ${data_feats}/org/${dset}/{segments,wav.scp,reco2file_and_channel}
+            _opts=
+            if [ -e data/"${dset}"/segments ]; then
+                # "segments" is used for splitting wav files which are written in "wav".scp
+                # into utterances. The file format of segments:
+                #   <segment_id> <record_id> <start_time> <end_time>
+                #   "e.g. call-861225-A-0050-0065 call-861225-A 5.0 6.5"
+                # Where the time is written in seconds.
+                _opts+="--segments data/${dset}/segments "
+            fi
+
+            
+            _spk_list=" "
+            for i in $(seq ${spk_num}); do
+                _spk_list+="spk${i} "
+            done
+            if $use_noise_ref; then
+                # reference for denoising ("noise1 noise2 ... niose${noise_type_num} ")
+                _spk_list+=$(for n in $(seq $noise_type_num); do echo -n "noise$n "; done)
+            fi
+            if $use_dereverb_ref; then
+                # reference for dereverberation
+                _spk_list+="dereverb "
+            fi
+
+            for spk in ${_spk_list} "wav" ; do
+                # shellcheck disable=SC2086
+                scripts/audio/format_wav_scp.sh --nj "${nj}" --cmd "${train_cmd}" \
+                    --out-filename "${spk}.scp" \
+                    --audio-format "${audio_format}" --fs "${fs}" ${_opts} \
+                    "data/${dset}/${spk}.scp" "${data_feats}/org/${dset}" \
+                    "${data_feats}/org/${dset}/logs/${spk}" "${data_feats}/org/${dset}/data/${spk}"
+            done
+
+            echo "${feats_type}" > "${data_feats}/org/${dset}/feats_type"
+        done
+        
+    else
+        log "Error: not supported: --feats_type ${feats_type}"
+        exit 2
+    fi
+fi
+
+
+if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
+    log "Stage 4: Remove short data: ${data_feats}/org -> ${data_feats}"
+
+    for dset in "${train_set}" "${dev_set}" ${eval_sets}; do
```

Comment:
There are some changes in the recent PR (https://github.com/espnet/espnet/pull/2112. Basically we should not prune the eval set to reproduce the same results with the others). Please reflect them.

Reply:
Done.